src/components/pager.tsx

```diff
@@ -8,14 +8,15 @@ import { Link, useHistory } from 'react-router-dom'
  * for pages a user would navigate through. This way the browser back
  * button can behave as expected based on the user's actual sequence of actions,
  * but a previous / next flow can be available as well for a sequence of screens.
- * @param pageUrls a list of urls, starting with '/' (history.location.pathname)
- * @returns [goToPreviousPage, goToNextPage, previousUrl, currentUrl]
+ * @param pages a list of pages A
+ * @param url gets a url out of a page A, starting with '/' (history.location.pathname)
+ * @returns [previousPage, goToNextPage]
  */
-export const usePager = (pageUrls: string[]): [() => void, () => void, string | undefined, string] => {
+export const usePager = <A, >(pages: A[], url: (a: A) => string): [A | undefined, (() => void) | undefined] => {
   const history = useHistory()
 
   const navPage = (path: string): number | undefined => {
-    const found = pageUrls.findIndex(p => p === path)
```

Comment:
Doesn't have to be done in this pull request, but maybe the PageUrls could be changed to a dictionary with the path to index mapping so that the list of strings is not looped through every time. Something like

`pageIndex = pageUrls[path]`

That way also this navPage function would not be needed.

Reply:
I'll think about it. Dictionary key order is not maintained though, so I would have to use some other data structure that combines array indexing with hash lookup.

Reply:
I was thinking the corresponding value of the path key would be the array index

Reply:
Take one more look, I went ahead and found an optimization. Just using Map didn't quite work - it remembers the order of the keys but only gives an iterator. It's not indexable by key insertion order, only iterable.

Reply:
Looks great! Thank you